[Bug]: TLSX_SetResponse ifdef-Preprocessor-directives not sufficient

### Contact Details

_No response_

### Version

5.5.4

### Description

If none of the several modules that use TLSX_SetResponse are enabled an error is shown at compile-time. The `#ifdef` preprocessor-directive is not sufficient and additional checks are needed there. For example `HAVE_ALPN`, `HAVE_SNI`, `HAVE_TRUSTED_CA` are guarding the usages, if none of those are defined the function ends up being defined but not used.

### Reproduction steps

Try to compile the example using the PR from https://github.com/RIOT-OS/RIOT/pull/19302

### Relevant log output

```shell
wolfssl/src/tls.c:1411:13: error: ‘TLSX_SetResponse’ defined but not used [-Werror=unused-function]
 1411 | static void TLSX_SetResponse(WOLFSSL* ssl, TLSX_Type type)
      |             ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/src/tls.c
+++ b/src/tls.c
@@ -1412,7 +1412,13 @@ int TLSX_HandleUnsupportedExtension(WOLFSSL* ssl)
 
 #endif
 
-#if !defined(NO_WOLFSSL_SERVER) || defined(WOLFSSL_TLS13)
+#if defined(WOLFSSL_TLS13) || (!defined(NO_WOLFSSL_SERVER) \
+    && (defined(WOLFSSL_SRTP) || defined(HAVE_SESSION_TICKET) \
+    || defined(HAVE_SECURE_RENEGOTIATION) || defined(HAVE_SERVER_RENEGOTIATION_INFO) \
+    || defined(HAVE_SUPPORTED_CURVES) || defined(HAVE_CERTIFICATE_STATUS_REQUEST_V2) \
+    || defined(HAVE_CERTIFICATE_STATUS_REQUEST) || defined(HAVE_TRUNCATED_HMAC) \
+    || defined(HAVE_MAX_FRAGMENT) || defined(HAVE_TRUSTED_CA) || defined(HAVE_SNI) \
+    || defined(HAVE_ALPN)))
 /** Mark an extension to be sent back to the client. */
 static void TLSX_SetResponse(WOLFSSL* ssl, TLSX_Type type)
 {

--- a/wolfcrypt/src/port/pic32/pic32mz-crypt.c
+++ b/wolfcrypt/src/port/pic32/pic32mz-crypt.c
@@ -89,11 +89,13 @@ static int Pic32Crypto(const byte* pIn, int inLen, word32* pOut, int outLen,
     bufferDescriptor *bd_p;
     byte *in_p;
     byte *out_p;
+    unsigned int *updptrVal_p;
     word32* dst;
     word32 padRemain;
     int timeout = 0xFFFFFF;
     word32* in = (word32*)pIn;
     word32* out = pOut;
+    word32 updptrVal = 0;
     int isDynamic = 0;
 
     /* check args */
@@ -125,6 +127,7 @@ static int Pic32Crypto(const byte* pIn, int inLen, word32* pOut, int outLen,
     bd_p = KVA0_TO_KVA1(&bd);
     out_p= KVA0_TO_KVA1(out);
     in_p = KVA0_TO_KVA1(in);
+    updptrVal_p = KVA0_TO_KVA1(&updptrVal);
 
     /* Sync cache if in physical memory (not flash) */
     if (PIC32MZ_IF_RAM(in_p)) {
@@ -193,6 +196,7 @@ static int Pic32Crypto(const byte* pIn, int inLen, word32* pOut, int outLen,
         if (in != out)
             XMEMSET(out_p, 0, outLen); /* clear output buffer */
         bd_p->DSTADDR = (unsigned int)KVA_TO_PA(out);
+        bd_p->UPDPTR = (unsigned int)KVA_TO_PA(updptrVal_p);
     }
     else {
         /* hashing */

